ebos/nncsorter.cpp

```diff
@@ -0,0 +1,89 @@
+// -*- mode: C++; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*-
+// vi: set et ts=4 sw=4 sts=4:
+/*
+  This file is part of the Open Porous Media project (OPM).
+
+  OPM is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 2 of the License, or
+  (at your option) any later version.
+
+  OPM is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+
+  Consult the COPYING file in the top-level source directory of this
+  module for the precise wording of the license and the list of
+  copyright holders.
+*/
+
+#include <opm/common/OpmLog/OpmLog.hpp>
+#include <opm/parser/eclipse/EclipseState/Grid/NNC.hpp>
+
+#include <sstream>
+#include <vector>
+#include <algorithm>
+#include <iostream>
+namespace Ewoms
+{
+namespace Detail
```

Comment:
please remove the `Detail` namespace: nested namespaces should be avoided in opm-material, ewoms and core-ebos. (as should deeply nested locations of include files.)

Reply:
Find by me but it exposes the implementation details ebos.
What the heck is core-ebos? This is opm-simulators!

Reply:
core-ebos is everyting under the `ebos` directory.

Reply:
also: the fact that it is an implementation detail is -- in my not so humble opinion -- made abundantly clear by the function name...